### DISCUSSION TITLE
Confusion about PSR12 ruleset in this repo: Are PSR12 sniffs actually included?

### DISCUSSION THREAD

**Discussion by finn-matti** [squizlabs/PHP_CodeSniffer#3865] [Q&A] [answered]
Take a look at, e.g.: https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/PSR12/ruleset.xml#L256

```
<!-- Expressions in parentheses MAY be split across multiple lines, where each subsequent line is indented at least once. When doing so, the first condition MUST be on the next line. The closing parenthesis and opening brace MUST be placed together on their own line with one space between them. Boolean operators between conditions MUST always be at the beginning or at the end of the line, not a mix of both. -->
    <!-- checked by PSR12.ControlStructures.ControlStructureSpacing -->
    <!-- checked by PSR12.ControlStructures.BooleanOperatorPlacement -->
    <!-- checked by Squiz.ControlStructures.ControlSignature -->
```

This kind of construct is encountered in other places with other rules, too, but the difference to for example https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/PSR12/ruleset.xml#L250 is the slightly different wording and that the mentioned sniff is actually included above:

```
<!-- 5.1 if, elseif, else -->

    <!-- else and elseif are on the same line as the closing brace from the earlier body. -->
    <!-- checked by Squiz.ControlStructures.ControlSignature included above -->
```

I do not see the inclusion of the psr12 sniffs. And I also have not found any kind of reference to those rules in a different way. Is it just by convention that sniffs that lay bellow a ruleset.xml-file in a subdir called Sniffs are included, or what magic includes them?

**Comment by jrfnl** [comment, marked as answer]
> Is it just by convention that sniffs that lay bellow a ruleset.xml-file in a subdir called Sniffs are included, or what magic includes them?

Yes, that's by design.
A `ruleset.xml` file for a standard will automatically include all sniffs defined in that standard.
